packages/compiler/src/source.ts

```diff
@@ -1,31 +1,92 @@
 import type { BOSModule } from '@bos-web-engine/common';
 import {
+  BLOCK_HEIGHT_KEY,
   SOCIAL_COMPONENT_NAMESPACE,
   SocialDb,
+  SocialGetParams,
 } from '@bos-web-engine/social-db';
 
-import { ComponentEntry } from './types';
+import {
+  ComponentEntryWithBlockHeight,
+  ComponentSourcesResponse,
+  SocialComponentWithBlockHeight,
+  SocialComponentsByAuthor,
+  SocialComponentsByAuthorWithBlockHeight,
+  SocialWidgetWithBlockHeight,
+} from './types';
+
+function prepareSource(response: SocialComponentsByAuthor): {
+  [key: string]: BOSModule;
+} {
+  return Object.entries(response).reduce(
+    (sources, [author, { [SOCIAL_COMPONENT_NAMESPACE]: componentEntry }]) => {
+      Object.entries(componentEntry).forEach(([componentName, component]) => {
+        if (component) {
+          const { '': source, css } = component;
+          sources[`${author}/${componentName}`] = {
+            component: source,
+            css,
+          };
+        } else {
+          console.error(`Invalid component source: ${component}`);
+        }
+      });
+      return sources;
+    },
+    {} as { [key: string]: BOSModule }
+  );
+}
+
+function isNotABlockEntry<T>(
+  entryKey: string,
+  entryValue: any
+): entryValue is T {
+  return typeof entryValue !== 'number' && entryKey !== BLOCK_HEIGHT_KEY;
+}
+
+function prepareSourceWithBlockHeight(
+  response: SocialComponentsByAuthorWithBlockHeight
+) {
+  return Object.entries(response).reduce((sources, [entryKey, entryValue]) => {
+    if (
+      isNotABlockEntry<SocialComponentWithBlockHeight>(entryKey, entryValue)
+    ) {
+      const { [SOCIAL_COMPONENT_NAMESPACE]: component } = entryValue;
+
+      if (
+        isNotABlockEntry<SocialWidgetWithBlockHeight>(
+          SOCIAL_COMPONENT_NAMESPACE,
+          component
+        )
+      ) {
+        Object.entries(component).forEach(([componentKey, componentValue]) => {
+          if (
+            isNotABlockEntry<ComponentEntryWithBlockHeight>(
+              componentKey,
+              componentValue
+            )
+          ) {
+            const sourceKey = `${entryKey}/${componentKey}`;
+            sources[sourceKey] = {
+              component: componentValue[''][''],
+              css: componentValue.css[''],
+              blockHeight: componentValue[BLOCK_HEIGHT_KEY],
+            };
+          }
+        });
+      }
+    }
+
+    return sources;
+  }, {} as ComponentSourcesResponse);
+}
 
 export async function fetchComponentSources(
```

Comment:
By convention, most of the code in this repo uses named parameters once there are more than one or two parameters. This to avoid [connascence of position](https://connascence.io/position.html) and makes usage more clear overall